src/microbe_stage/editor/PatchMapDrawer.cs

```diff
@@ -107,27 +128,391 @@ public override void _Draw()
         if (Map == null)
             return;
 
+        // Create connections between regions if they dont exist.
+        // Also center the scroll to the player patch
+        if (connections.Count == 0)
+        {
+            CenterScroll();
+            CreateRegionLinks();
+        }
+
+        DrawRegionLinks();
+        DrawRegions();
+
+        // For special regions draw the connection between them and the normal region.
+        foreach (var entry in Map.SpecialRegions)
+        {
+            var region = entry.Value;
+            var patch = region.Patches.First();
+            var adjacent = patch.Adjacent.First();
+            var adjacentRegion = adjacent.Region;
+            var start = Center(patch.ScreenCoordinates);
+            var end = Center(adjacent.ScreenCoordinates);
+
+            var regionRect = new Rect2(region.ScreenCoordinates, new Vector2(region.Width, region.Height));
+            var adjacentRect = new Rect2(adjacentRegion.ScreenCoordinates,
+                new Vector2(adjacentRegion.Width, adjacentRegion.Height));
+
+            // We get the intersection of the connections with the 2 adjacent regions (special and normal)
+            start = LineRectangleIntersection(start, end, regionRect);
+            end = LineRectangleIntersection(start, end, adjacentRect);
+
+            DrawNodeLink(start, end);
+            DrawRect(new Rect2(region.ScreenCoordinates, new Vector2(region.Width, region.Height)),
+                new Color(0f, 0.7f, 0.5f, 0.7f), false, RegionLineWidth);
+        }
+
         // This ends up drawing duplicates but that doesn't seem problematic ATM
         foreach (var entry in Map.Patches)
         {
             foreach (var adjacent in entry.Value.Adjacent)
             {
-                var start = Center(entry.Value.ScreenCoordinates);
-                var end = Center(adjacent.ScreenCoordinates);
-
-                DrawNodeLink(start, end);
+                // Only draw connections if we haven't set regions (classic map) or patches belong to the same region
+                if (entry.Value.Region?.Name.Equals(adjacent.Region.Name) != false)
+                {
+                    var start = Center(entry.Value.ScreenCoordinates);
+                    var end = Center(adjacent.ScreenCoordinates);
+
+                    DrawNodeLink(start, end);
+                }
             }
         }
     }
 
+    public void MarkDirty()
+    {
+        dirty = true;
+    }
+
+    public void CenterScroll()
+    {
+        var parent = (DraggableScrollContainer)GetParent();
+        var coords = PlayerPatch!.ScreenCoordinates - parent.GetRect().End / 2f;
+        parent.ScrollHorizontal = (int)coords.x;
+        parent.ScrollVertical = (int)coords.y;
+        parent.ResetZoom();
+    }
```

Comment:
This seems 100% like functionality that belongs in DraggableScrollContainer, this scene should just emit a signal when this functionality is wanted and the parent should then react.

Reply:
Fixed.